cparser/Elab.ml

```diff
@@ -1175,23 +1199,21 @@ let rec elab_designator loc env zi desig =
       | Some zi' ->
           elab_designator loc env zi' desig'
       | None ->
-          error loc "%s has no member named %s" (I.name zi) name;
+          error loc "field designator '%s' does not refer to any field in type '%s'" name (I.name zi);
           raise Exit
       end
   | ATINDEX_INIT a :: desig' ->
       let expr,env = (!elab_expr_f loc env a) in
       begin match Ceval.integer_expr env expr with
       | None ->
-          error loc "array element designator for %s is not a compile-time constant"
-                    (I.name zi);
+          error loc "expression is not an integer constant expression";
```

Comment:
The original message is more informative.  (Gives more context about which array is wrong.)